service/src/test/java/cz/muni/fi/pa165/mushrooms/service/ForestServiceImplTest.java

```diff
@@ -0,0 +1,348 @@
+
+            forestDao.findByName(anyString);
+            result = new Delegate() {
+                Forest foo(String name) {
+                    return database.findByName(name);
+                }
+            };
+            minTimes = 0;
+
+            forestDao.findAll();
+            result = new Delegate() {
+                List<Forest> foo() {
+                    return database.findAll();
+                }
+            };
+            minTimes = 0;
+
+        }};
+    }
+
+    @Test
+    public void findAllForests() {
+
+        assertThat(service.findAllForests()).isEmpty();
+
+        database.create(forest1);
+        database.create(forest2);
+        database.create(forest3);
+
+        assertThat(service.findAllForests()).containsExactlyInAnyOrder(forest1, forest2, forest3);
+    }
+
+    @Test
+    public void findForestById_valid() {
+        database.create(forest1);
+        database.create(forest2);
+
+        assertThat(service.findForestById(forest1.getId())).isEqualToComparingFieldByField(forest1);
+    }
+
+    @Test
+    public void findForestById_invalid() {
+        database.create(forest1);
+        database.create(forest2);
+
+        assertThat(service.findForestById(forest1.getId() + forest2.getId())).isNull();
+    }
+
+    @Test
+    public void findForestById_null() {
+        database.create(forest1);
+        database.create(forest2);
+
+        assertThatThrownBy(() -> service.findForestById(null)).isInstanceOf(DataAccessException.class);
+    }
+
+    @Test
+    public void FindForestByName_null() {
+
+        database.create(forest1);
+        database.create(forest2);
+
+        assertThatThrownBy(() -> service.findForestByName(null)).isInstanceOf(DataAccessException.class);
+    }
+
+    @Test
+    public void FindForestByName_nonexistent() {
+
+        database.create(forest1);
+        database.create(forest2);
+
+        assertThat(service.findForestByName(forest3.getName())).isNull();
+    }
+
+    @Test
+    public void FindForestByName_valid() {
+
+        database.create(forest1);
+        database.create(forest2);
+        database.create(forest3);
+
+        assertThat(service.findForestByName(forest3.getName())).isEqualToComparingFieldByField(forest3);
+    }
+
+    @Test
+    public void FindForestByName_emptyString() {
+
+        database.create(forest1);
+        database.create(forest3);
+
+        assertThat(service.findForestByName("")).isNull();
+    }
+
+    @Test
+    public void createForest_valid() {
+
+        assertThat(forest1.getId()).isNull();
+        service.createForest(forest1);
+        assertThat(forest1.getId()).isNotNull();
+        assertThat(database.findById(forest1.getId())).isEqualTo(forest1);
+    }
+
+    @Test
+    public void createForest_preexisting() {
+
+        forest2.setId(2L);
+        assertThatThrownBy(() -> service.createForest(forest2)).isInstanceOf(DataAccessException.class);
+    }
+
+    @Test
+    public void createForest_conflicting() {
```

Comment:
"conflicting" with which constraint? here, it's about creating an entity which already has an ID, which is tested in the previous method